lib/update/phase_coredns.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package update
+
+import (
+	"context"
+
+	"github.com/gravitational/gravity/lib/storage"
+	"github.com/gravitational/rigging"
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	corednsResourceName = "gravity:coredns"
+)
+
+func (r phaseBuilder) corednsPhase(leadMaster storage.Server) *phase {
+	phase := root(phase{
+		ID:          "coredns",
+		Description: "Provision coredns resources",
+		Executor:    coredns,
+		Data: &storage.OperationPhaseData{
+			Server: &leadMaster,
+		},
+	})
+	return &phase
+}
+
+type updatePhaseCoreDNS struct {
+	log.FieldLogger
+	kubernetesOperation
+}
+
+// NewPhaseCoreDNS create an upgrade phase to add coredns rbac permissions
+// HACK: This phase needs to run before updating planet, as coredns is embedded in planet
```

Comment:
"HACK" is not too useful, I would either just get rid of this word, or replace it with a proper TODO (or FIXME) that explains what needs to be done to unhack it.